### ISSUE TITLE
Issue with Support In-Game and Characterprofile.php

### ISSUE THREAD

**Issue by deviljin112** [Znote/ZnoteAAC#386]
Hello,

Firstly, Support In-Game. It displays GOD character at the very bottom but also spits out an error on the side :

```
Notice: Undefined offset: 0 in C:\xampp\htdocs\engine\function\users.php on line 142

Notice: Undefined offset: 1 in C:\xampp\htdocs\engine\function\users.php on line 142

Notice: Undefined offset: 2 in C:\xampp\htdocs\engine\function\users.php on line 142

Notice: Undefined offset: 3 in C:\xampp\htdocs\engine\function\users.php on line 142

Notice: Undefined offset: 4 in C:\xampp\htdocs\engine\function\users.php on line 142

Notice: Undefined offset: 5 in C:\xampp\htdocs\engine\function\users.php on line 142

Notice: Undefined index: group_id in C:\xampp\htdocs\support.php on line 16
```

But at the very bottom of the page, it shows the table with Group, Name and Status. Very weird.

Other issue is with CharacterProfile.php. I couldn't find any fix for it as it seems fine, DB has the correct tables etc and the code appears correct. Error:

```
string(98) "SELECT SUM(`value`) AS `sum` FROM `player_storage` WHERE `key` LIKE '30___' AND `player_id`=(int)7"
(query - SQL error)
Type: select_single (select single row from database)

You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near 'int)7' at line 1
```

I've tried playing around with it, but the error persists.

EDIT1: Running Master-branch of ZnoteACC as well as TFS1.3 if that is important

Thank you,
D3v

**Comment by Znote** [comment]
https://github.com/Znote/ZnoteAAC/blob/35e2ff7f43a3be8774db945d5a11c097120b651f/characterprofile.php#L27

Try to replace this line with:
```lua
 				$user_id = (int)$user_id; 
```

As for the support.php file, I confirm the same issue. You can disable warnings there for now while I look for a patch.

**Comment by deviljin112** [comment]
The characterprofile.php issue was with the layout i had, i had an old file within layout and it was causing glitches, i never realised until now. I have fixed that issue now. But the support.php persists.

I have disabled all warnings for support.php for now, looking forward to seeing a fix. I (not a php expert) thing the issue may lay with how it pulls the config('ingame_positions') in support.php. As it would appear that it iterates through every group and fails to find (in the databse) a coresponding account with this group therefore failing to index it correctly, as GOD is group 6 and that group is not  in the error. I again am not an expert, but in python and other languages this could be fixed with a "try:" function. Maybe that can be implemented here, where it "tries" to find that index in the DB if fails then continue else index as online. Rather than having an index error spit into the html. But thats just my theory, im not a web developer :D just a programmer.
